com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -1197,10 +1197,22 @@ internal void ShutdownInternal()
                 }
             }
 
+            // Unregister network updates before trying to disconnect the client
+            this.UnregisterAllNetworkUpdates();
+
             if (IsClient && IsListening)
```

Comment:
Wondering if this check needs to be:
`if (!IsServer && IsClient && IsListening)`
Or... the comment below needs to be updated.